subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -108,6 +195,54 @@ static int fp_gatt_rsp_notify(struct bt_conn *conn, const struct bt_gatt_attr *a
 	return err;
 }
 
+static int notify_personalized_name(struct bt_conn *conn)
+{
+	int err;
+	char pn[FP_STORAGE_PN_BUF_LEN];
+	size_t pn_len;
+	uint8_t nonce[FP_CRYPTO_ADDITIONAL_DATA_NONCE_LEN];
+	struct bt_gatt_attr *additional_data_attr;
+
+	err = fp_storage_pn_get(pn);
+	if (err) {
+		LOG_ERR("Failed to get Personalized Name from storage: err=%d", err);
+		return err;
+	}
+
+	pn_len = strlen(pn);
+	if (pn_len == 0) {
+		LOG_DBG("Personalized Name is empty");
+		return 0;
+	}
+
+	uint8_t packet[FP_CRYPTO_ADDITIONAL_DATA_HEADER_LEN + pn_len];
+
+	/* Generate random nonce. */
+	err = sys_csrand_get(nonce, sizeof(nonce));
+	if (err) {
+		LOG_ERR("Failed to generate nonce: err=%d", err);
+		return err;
+	}
+
+	err = fp_keys_additional_data_encode(conn, packet, pn, pn_len, nonce);
+	if (err) {
+		LOG_ERR("Failed to encode additional data: err=%d", err);
+		return err;
+	}
+
+	additional_data_attr = bt_gatt_find_by_uuid(fast_pair_svc.attrs, fast_pair_svc.attr_count,
+						    BT_UUID_FAST_PAIR_ADDITIONAL_DATA);
+	__ASSERT_NO_MSG(additional_data_attr != NULL);
+
+	err = bt_gatt_notify(conn, additional_data_attr, packet, sizeof(packet));
```

Comment:
What about checking CCCD values here, if `conn` will be different than NULL then notification will be send even if client doesn't enable it

Reply:
By default CONFIG_BT_GATT_ENFORCE_SUBSCRIPTION is enabled, so notification should not be send if client is not subscribed.